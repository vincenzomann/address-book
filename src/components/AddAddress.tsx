import React, { useState } from 'react';
import { RouteComponentProps } from '@reach/router';
import './sass/AddAddress.scss';
import Results from './Results';
import AddressForm from './AddressForm';
import { useContextProvider } from '../context/Context';
import AddressInfo from './AddressInfo';

interface Props extends RouteComponentProps { }

const Lookup: React.FC<Props> = () => {

	const { results, setResults, selected } = useContextProvider();

	const [error, setError] = useState('');
	const [message, setMessage] = useState('');
	// Determines which form to display
	const [mode, setMode] = useState<'search' | 'manual'>('search');

	return (
		<div id='addAddress'>
			<h1 className='pageTitle'>
				Add Address
			</h1>

			<div id='formContainer'>
				<AddressForm
					setResults={setResults}
					error={error} setError={setError}
					message={message} setMessage={setMessage}
					mode={mode} setMode={setMode}
				/>
			</div>

			{/* When user clicks on an address card it is displayed */}
			{selected.id &&
				<div className='selectedAddress'>
					Selected Address:
					<AddressInfo address={selected} />
				</div>
			}

			{/* Only display results on search mode */}
			<div id='results' style={{ display: mode === 'search' ? '' : 'none' }}>
				{results && <Results results={results} setError={setError} setMessage={setMessage} />}
			</div>
		</div>
	);
};

export default Lookup;
