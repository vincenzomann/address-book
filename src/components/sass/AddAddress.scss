#addAddress {
	@media only screen and (min-width: 700px) {
		padding: 1rem;
	}

	#pageTitle {
		font-weight: bold;
	}

	#formContainer {
		display: flex;
		justify-content: center;
		box-sizing: border-box;
		max-width: 800px;
		margin: 2rem auto;

		#switch {
			:hover {
				cursor: pointer;
			}
			.option {
				margin-right: 1rem;
				font-size: large;
			}
			.selected {
				border-bottom: 3px solid #009bc1;
				padding-bottom: 0.25rem;
			}
		}

		#form {
			display: flex;
			justify-content: center;
			flex-direction: column;
			.inputs {
				padding: 1rem 0;
				input {
					height: 2.5rem;
					width: 100%;
					margin: 0.5rem 0;
					border: 1px solid lightGrey;
					border-radius: 10px;
					padding: 1rem;
				}
			}

			#select {
				margin: 0.5rem 0;
				font-size: small;

				input {
					height: 100%;
				}
			}

			.error {
				width: 100%;
				font-size: 18px;
				color: #f36f5e;
			}
			.message {
				width: 100%;
				font-size: 18px;
				color: lightgreen;
				margin: 0.5rem;
			}

			#searchBtn {
				height: 48px;
				width: 100%;
				font-size: 18px;
				color: #009bc1;
				background: #e5f5f8;
				font-weight: bold;
				border: none;
				border-radius: 10px;
				display: flex;
				justify-content: center;
				align-items: center;
				cursor: pointer;
			}
		}
	}

	#results {
		margin: 1rem;
		:hover {
			color: #009bc1;
		}
		.result {
			margin: 1rem 0;
			cursor: pointer;
		}
		.card {
			padding: 1rem;
		}
		.added {
			border-color: lightgreen;
			color: lightgreen;
			:hover {
				color: lightgreen;
			}
		}
	}
}
