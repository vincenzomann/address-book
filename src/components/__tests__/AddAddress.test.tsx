import React from 'react';
import { shallow } from 'enzyme';
import AddAddress from '../AddAddress';

test('renders the component', () => {
	const component = shallow(<AddAddress />);
	expect(component).toMatchSnapshot();
});
