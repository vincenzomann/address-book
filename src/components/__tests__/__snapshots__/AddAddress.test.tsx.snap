// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  id="addAddress"
>
  <h1
    className="pageTitle"
  >
    Add Address
  </h1>
  <div
    id="formContainer"
  >
    <AddressForm
      error=""
      message=""
      mode="search"
      setError={[Function]}
      setMessage={[Function]}
      setMode={[Function]}
      setResults={[Function]}
    />
  </div>
  <div
    id="results"
    style={
      Object {
        "display": "",
      }
    }
  >
    <Results
      results={
        Object {
          "addresses": Array [],
          "latitude": 0,
          "longitude": 0,
          "postcode": "",
        }
      }
      setError={[Function]}
      setMessage={[Function]}
    />
  </div>
</div>
`;
